### PR TITLE
fix(import): Better handling of "Patch is empty"

### LINKED ISSUE #2351: lerna import fails "Command failed: git am -3 --keep-non-patch"
Hi,

From my lerna monorepo I'm trying to import an external project.

## Expected Behavior
`lerna import ../external_repo --flatten` should succeed

## Current Behavior
`lerna import ../external_repo --flatten` fails with the following error message:
```
~/D/o/lerna ❯❯❯ lerna import ../external_repo --flatten
info cli using local version of lerna
lerna notice cli v3.18.3
lerna info versioning independent
lerna info About to import 615 commits from ../external_repo into packages/external_repo
? Are you sure you want to import these commits onto the current branch? Yes
lerna info 1fa128eb 
...
lerna info 20db6d0e 
lerna ERR! import Rolling back to previous HEAD (commit 412f88d993e98e51b3242c61ff899ece8dbe341e)
lerna ERR! EIMPORT Failed to apply commit 20db6d0e.
lerna ERR! EIMPORT Command failed: git am -3 --keep-non-patch
lerna ERR! EIMPORT The current patch is empty.
lerna ERR! EIMPORT Quand vous avez résolu ce problème, lancez "git am --continue".
lerna ERR! EIMPORT Si vous préférez plutôt sauter ce patch, lancez "git am --skip".
lerna ERR! EIMPORT Pour restaurer la branche originale et arrêter de patcher, lancez "git am --abort".
lerna ERR! EIMPORT 
lerna ERR! EIMPORT 
lerna ERR! EIMPORT You may try again with --flatten to import flat history.
```
We can see lerna fails to apply the commit `20db6d0e`.
Here you can see the commit's info from the external repo
```
~/D/o/external_repo ❯❯❯ git show 20db6d0e
commit 20db6d0e6af4c8e59b380d1d9fac4054108517a5
Merge: 9ba5c7e6 e146b579
Author: xxxx <xxxx@gmail.com>
Date:   Tue Sep 17 16:46:18 2019 +0200

    Merge pull request #300 from xxxx/fix/xx
    
    Fix error value of null
```
and
```
~/D/o/external_repo ❯❯❯ git show 9ba5c7e6
commit 9ba5c7e6c4f06a2e6613c4ff611d180efa92cdb2
Merge: 933211d9 b889fafb
Author: xxx <xxx@gmail.com>
Date:   Tue Sep 17 14:17:23 2019 +0200

    Merge pull request #301 from xxx/hotFix/owp
    
    Fix Logarithm value CummulatedPerf and yesterday of RefPortfolio

~/D/o/external_repo ❯❯❯ git show e146b579
commit e146b579a55b5404658f55c74784a63e8b962396
Author: xxx <xxx@gmail.com>
Date:   Wed Sep 11 18:09:51 2019 +0200

    Fix error value of null

diff --git xxxx.js
index a212efea..2340eee0 100644
--- xxxx.js
+++ xxxx.js
...some diff details in code
```

## Possible Solution
I'm not sure what cause this problem and I have no idea how to make the import succeed.

<details><summary>lerna.json</summary><p>

```json
{                         
  "packages": [
    "packages/*"
  ],
  "npmClient": "yarn",
  "useWorkspaces": true,
  "version": "independent"
}
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

the error does not produces logs

</p></details>

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.18.3 |
| `npm --version`  | 6.13.0 |
| `yarn --version` | 1.19.1 |
| `node --version` | v10.17.0 |
| git | 2.24.0 |

| OS | Version |
| --- | --- |
| Arch Linux | 5.3.11-arch1-1 |

### LINKED ISSUE #2587: lerna import fails on Empty patch with `LANG=ru_RU.UTF-8`
I tried to import package into empty lerna monorepo but it was failing every time.
Command: `lerna import --flatten <path/to/repo>`
Output:
```
lerna ERR! import Rolling back to previous HEAD (commit fff5b779833c57820df2f76c812fd0e89641de07)
lerna ERR! EIMPORT Failed to apply commit ef18c6c.
lerna ERR! EIMPORT Command failed: git am -3 --keep-non-patch
lerna ERR! EIMPORT Патч пустой.
lerna ERR! EIMPORT Когда вы устраните эту проблему, запустите «git am --continue».
lerna ERR! EIMPORT Если вы хотите пропустить этот патч, то запустите «git am --skip».
lerna ERR! EIMPORT Чтобы вернуться на предыдущую ветку и остановить применение изменений, запустите «git am --abort».
lerna ERR! EIMPORT 
lerna ERR! EIMPORT 
lerna ERR! EIMPORT You may try again with --flatten to import flat history.
```

I spent couple of hours trying to understand and solve this and figured out that the problem is I had environment variable `LANG` set to `ru_RU.UTF-8` (this is the default value for macOS in Russia).
I stumbled upon this PR which seems to be fixing this same issue: https://github.com/lerna/lerna/pull/1059
However it fixes it only for users with English language. If I switch my local `LANG` environment to `en_US.UTF-8` then import works successfully.
While I understand that maybe it's not a problem with lerna but I feel like forcing users to switch their environment to use all features of lerna is wrong.
Plus fixing it will probably save others some time.

## Expected Behavior
Repo should be imported successfully.

## Current Behavior
Import fails with error.

## Possible Solution
I can think of two ways to solve this:
1. Set `LANG` to `en_US.UTF-8` automatically before running git commands.
2. Update `import` command to check for "Empty patch." message in languages other than English.

## Steps to Reproduce (for bugs)
1. Run `export LANG=ru_RU.UTF-8`
2. Try `lerna import --flatten` on any repo that contains empty commit.

<details><summary>lerna.json</summary><p>

```json
{
  "name": "affluent",
  "packages": [
    "packages/*"
  ],
  "private": true,
  "version": "0.0.0"
}
```
</p></details>

## Context
Import the repo into empty lerna project.

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.21.0 |
| `npm --version`  | 6.14.5 |
| `yarn --version` | 1.22.0 |
| `node --version` | v12.16.1 |

| OS | Version |
| --- | --- |
| macOS Catalina | 10.15.4 |

### PR DESCRIPTION
**Pull request by TimShilov**
## Description
Removed check for hard-coded "Patch is empty." git message in favor of `git diff COMMIT^!` check.
Fixes #2351 , fixes #2587

## Motivation and Context
Command `lerna import --flatten <path/to/repo>` fails if `LANG` environment variable set to `ru_RU.UTF-8` (default value on macOS in Russia).
Example output:
```
lerna ERR! import Rolling back to previous HEAD (commit fff5b779833c57820df2f76c812fd0e89641de07)
lerna ERR! EIMPORT Failed to apply commit ef18c6c.
lerna ERR! EIMPORT Command failed: git am -3 --keep-non-patch
lerna ERR! EIMPORT Патч пустой.
lerna ERR! EIMPORT Когда вы устраните эту проблему, запустите «git am --continue».
lerna ERR! EIMPORT Если вы хотите пропустить этот патч, то запустите «git am --skip».
lerna ERR! EIMPORT Чтобы вернуться на предыдущую ветку и остановить применение изменений, запустите «git am --abort».
lerna ERR! EIMPORT 
lerna ERR! EIMPORT 
lerna ERR! EIMPORT You may try again with --flatten to import flat history.
```

## How Has This Been Tested?
Updated command's code and successfully imported repo with multiple locales including: `en_US.UTF-8`, `ru_RU.UTF-8`, `en_IN.utf8`, `es_ES.UTF-8` and `de_CH.ISO8859-1`.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: commands/import/index.js [hunk 1]
```diff
@@ -237,7 +237,9 @@ class ImportCommand extends Command {
           tracker.completeWork(1);
         })
         .catch(err => {
-          if (err.stdout.indexOf("Patch is empty.") === 0) {
+          const patchIsEmptyMessages = ["Patch is empty.", "Патч пустой."];
```
Comment (evocateur) [inline commands/import/index.js:240]: Is there any other way of detecting this status _independently_ of an arbitrary localized string?

Comment (TimShilov) [reply commands/import/index.js]: I found a better way that works without hard-coded list of strings.

**Comment by TimShilov** [comment]
I updated the code to use git diff COMMIT^! command to check if the actual commit is empty.

Checked that it works on multiple locales including:  en_US.UTF-8, ru_RU.UTF-8, en_IN.utf8, es_ES.UTF-8
 and de_CH.ISO8859-1.

**Comment by TimShilov** [comment]
@evocateur I reviewed your request change and updated code long time ago. Have you got a chance to review it?

File: commands/import/index.js (hunk 1 above)
Comment (TimShilov) [reply commands/import/index.js]: @evocateur I reviewed your request change and updated code long time ago. Is there anything else I should do?

**Comment by TimShilov** [comment]
@evocateur Hi! I fixed your change request.
Is there anything I can/need to do to get this merged?

**Review by evocateur — APPROVED** [review]
Thanks!

**Pull request merged**